electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -251,6 +251,28 @@ def merge_production_breakdowns(
             production_breakdowns.events.append(prod)
         return production_breakdowns
 
+    @staticmethod
+    def update_production_breakdowns(
+        production_breakdowns: "ProductionBreakdownList",
+        new_production_breakdowns: "ProductionBreakdownList",
+        logger: Logger,
+    ) -> "ProductionBreakdownList":
+        """Given a new batch of production breakdowns, update the existing ones."""
+        if len(new_production_breakdowns) == 0:
+            return production_breakdowns
+        elif len(production_breakdowns) == 0:
+            return new_production_breakdowns
+
+        for event in new_production_breakdowns.events:
```

Comment:
I'd like to flag that at the medium level we're iterating through `production_breakdowns.events` and if we enter the conditional we update some elements within that. 

It's not ideal. I understand that the datetimes match and that it shouldn't pose a problem in this case, but to increase robustness, could we separate concerns? 

A way could be to create a variable beforehand with all datetimes of existing events. This way you can remove the iteration through `production_breakdowns.events`

Reply:
Something like this?
```python
        existing_events = {
            event.datetime: event for event in production_breakdowns.events
        }

        for new_event in new_production_breakdowns.events:
            if new_event.datetime in existing_events:
                existing_event = existing_events[new_event.datetime]
                existing_event = ProductionBreakdown.update(existing_event, new_event)
```

Reply:
So creating a say list of the existing datetimes and then check if the new datetime is in that dict and then match it using that?

Reply:
Exactly! Looks great :)